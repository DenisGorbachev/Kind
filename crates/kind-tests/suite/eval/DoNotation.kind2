Main : Maybe U60
Main =
    do Maybe {
        Maybe.some 3
        Maybe.pure 2
        ask res  = Maybe.pure 2
        ask res2 = Maybe.pure 3
        match Maybe (Maybe.some 4) {
            some val => Maybe.pure (+ 1000 (+ val (+ res res2)))
            none     => Maybe.none
        }
    }