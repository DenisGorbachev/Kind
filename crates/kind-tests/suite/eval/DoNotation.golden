(Maybe.some 1009)